### PR TITLE
Wildcard intervals

### LINKED ISSUE #43198: Add wildcard and prefix intervals
We should expose the `wildcard` and `prefix` IntervalSource implementations
as providers in elasticsearch.

```
{ "prefix" : {
    "term" : term prefix,
    "analyzer" : optional analyzer (uses normalize() function),
    "use_field" : optional field switch
}}
{ "wildcard" : {
    "term" : term glob,
    "analyzer" : optional analyzer (uses normalize() function),
    "use_field" : optional field switch
}}
```

Span queries also allow you to wrap regexp and term range queries, and we can
consider adding these here as well, although they will also require some work in
lucene.

The `prefix` source can piggy-back on text fields `index_prefixes` functionality to
rewrite to simple term intervals sources

### PR DESCRIPTION
**Pull request by romseygeek**
This commit adds a `wildcard` intervals source, similar to the `prefix`.  It
also changes the `term` parameter in `prefix` to read `prefix`, to bring it
in to line with the `pattern` parameter in `wildcard`.

Closes #43198

### CODE REVIEW

**Comment by romseygeek** [comment]
@elasticmachine run elasticsearch-ci/default-distro

File: docs/reference/query-dsl/intervals-query.asciidoc [hunk 1]
```diff
@@ -101,6 +101,24 @@ If specified, then match intervals from this field rather than the top-level fie
...
+[[intervals-wildcard]]
+==== `wildcard`
+
+The `wildcard` rule finds terms that match a wildcard pattern.  The pattern will
+expand to match at most 128 terms; if there are more matching terms in the index,
+then an error will be returned.
+
+[horizontal]
+`pattern`::
+Match terms starting matching this pattern
```
Comment (jimczi) [inline docs/reference/query-dsl/intervals-query.asciidoc]: Can you add the operators that are supported (`?` and `*` ?) and maybe a warning regarding leading wildcards ?

**Review by jimczi — APPROVED** [review]
code LGTM, I left a comment regarding the docs

**Comment by romseygeek** [comment]
@elasticmachine update branch

**Pull request merged**